onnxruntime/core/framework/orttraining_partial_executor.cc

```diff
@@ -233,6 +233,22 @@ Status PartialExecutor::Execute(const SessionState& session_state, const std::ve
     // construct OpKernelContext
     // TODO: log kernel inputs?
     OpKernelContextInternal op_kernel_context(session_state, frame, *p_op_kernel, logger, false);
+
+    // Cache lookup. Currently we only cache single-output nodes,
+    // to keep memory overhead impact in check. Hence we only look in cache
+    // if the current node has one output.
+    bool reuse_cached_value = false;
+    std::string cached_arg_name;
+    if (check_cache_) {
```

Comment:
>check_cache_) [](http://example.com/codeflow?start=8&length=13)

Please use cache_.empty(), it complexity is constant: [https://www.cplusplus.com/reference/unordered_map/unordered_map/empty/](https://www.cplusplus.com/reference/unordered_map/unordered_map/empty/)

Reply:
see https://github.com/microsoft/onnxruntime/pull/8273#issuecomment-900515827